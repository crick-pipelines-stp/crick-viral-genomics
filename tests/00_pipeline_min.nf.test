"""
Test an aboslute minimal run to check for syntax errors etc.

[-        ] process > MERGE_REFS                                                -
[53/d5d5f9] process > SAMPLESHEET_CHECK (test_nanopore_cov_amp_min.csv)         [100%] 1 of 1 ✔
[-        ] process > CAT_FASTQ                                                 -
[82/3605c3] process > SAMTOOLS_FAIDX (SARS-CoV-2.fasta)                         [100%] 1 of 1 ✔
[-        ] process > SAMTOOLS_INDEX_VIRUS                                      -
[-        ] process > SAMTOOLS_FILTER_VIRUS                                     -
[-        ] process > BAM_VIRAL_SORT_STATS:SAMTOOLS_SORT                        -
[-        ] process > BAM_VIRAL_SORT_STATS:SAMTOOLS_INDEX                       -
[-        ] process > BAM_VIRAL_SORT_STATS:BAM_STATS_SAMTOOLS:SAMTOOLS_STATS    -
[-        ] process > BAM_VIRAL_SORT_STATS:BAM_STATS_SAMTOOLS:SAMTOOLS_FLAGSTAT -
[-        ] process > BAM_VIRAL_SORT_STATS:BAM_STATS_SAMTOOLS:SAMTOOLS_IDXSTATS -
[-        ] process > SEQKIT_BAM                                                -
[-        ] process > SAMTOOLS_MPILEUP                                          -
[-        ] process > GEN_COUNT_TABLE                                           -
[-        ] process > MOSDEPTH                                                  -
"""

nextflow_pipeline {
    name "Test minimal end-to-end pipeline"
    script "../main.nf"
    tag "pipeline"

    test("pipeline_min") {
        tag "pipeline/min"
        when {
            params {
                outdir        = "$outputDir"
                viral_fasta   = "$projectDir/data/refs/covid/SARS-CoV-2.fasta"
                samplesheet   = "$projectDir/tests/data/samplesheets/test_nanopore_cov_amp_min.csv"
                run_reporting = false
            }
        }
        then {
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.trace.succeeded().size(),
                    stable_name,
                    stable_path
                ).match() }
            )
        }
    }
}
