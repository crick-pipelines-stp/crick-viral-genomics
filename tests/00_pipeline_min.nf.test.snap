{
    "pipeline_min": {
        "content": [
            2,
            [
                "00_reference",
                "00_reference/SARS-CoV-2.fasta.fai",
                "pipeline_info"
            ],
            [
                "SARS-CoV-2.fasta.fai:md5,6806a14930eab7e2de4e9776d79bcab2"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-14T11:10:27.262795"
    }
}